core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2003,7 +2003,7 @@ sealed trait ZIO[-R, +E, +A]
    */
   final def tapSomeError[R1 <: R, E1 >: E, A1 >: A](
     f: PartialFunction[E, ZIO[R1, E1, A1]]
-  )(implicit ev: CanFail[E], trace: Trace): ZIO[R1, E1, A1] =
+  )(implicit ev: CanFail[E], trace: Trace): ZIO[R1, E1, A] =
```

Comment:
It looks to me like `f` passed to `tapSomeError` does not need to take a parameter `A1` and can simply read like:

```scala
  final def tapSomeError[R1 <: R, E1 >: E](
    f: PartialFunction[E, ZIO[R1, E1, Any]]
  )(implicit ev: CanFail[E], trace: Trace): ZIO[R1, E1, A1] =
```

Reply:
@jdegoes Done in #7222.